wgsl/index.bs

```diff
@@ -7103,32 +7103,73 @@ The second phase explores that graph, resulting in either rejecting the program,
         A consequence of this interpretation is that every node reachable from RequiredToBeUniform corresponds to something which must be uniform for the program to be valid, and every node from which cannotBeUniform is reachable corresponds to something whose uniformity we cannot guarantee. It follows that we have a uniformity violation (and thus reject the program) if there is any path from RequiredToBeUniform to MayBeNonUniform.
 </div>
 
-In more detail, here is what we compute for each function:
+For each function, two tags are computed:
+  * A <dfn noexport>call site tag</dfn> describing the control flow uniformity requirements on the [=call sites=] of the function, and
+  * A <dfn noexport>function tag</dfn> describing the function's effects on uniformity.
 
-* A tag about the control-flow in which the function can be called: one of {AlwaysRequiredToBeUniform, RequiredToBeUniformForControlFlowAfter}
-* A tag for each parameter of the function: one of {AlwaysRequiredToBeUniform, RequiredToBeUniformForControlFlowAfter, RequiredToBeUniformForReturnValue, NoRestriction}
-* A tag for the function as a whole: one of {ControlFlowMayBeNonUniformAfter, ReturnValueMayBeNonUniform, NoRestriction}
+Additionally, for each [=formal parameter=] of a function, a <dfn
+noexport>parameter tag</dfn> is computed that describes the uniformity
+requirement of the parameter value.
 
-Here is the algorithm for computing this data for a given function:
+<table class='data'>
+  <caption>[=Call site tag=] values</caption>
+  <thead>
+    <tr><th>Call Site Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>CallSiteRequiredToBeUniform</dfn>
+      <td>The function must only be called from [=uniform control flow=].
+  <tr><td><dfn noexport>CallSiteNoRestriction</dfn>
+      <td>The function may be called from [=uniform control flow|non-uniform control flow=].
+</table>
+
+<table class='data'>
+  <caption>[=Function tag=] values</caption>
+  <thead>
+    <tr><th>Function Tag<th>Description
+  </thead>
+  <tr><td><dfn noexport>SubsequentControlFlowMayBeNonUniform</dfn>
```

Comment:
The "May" here is confusing about which entity is being constrained.
We're trying to say that executing the function may cause subsequent control flow to be nonuniform.
An alternative (to a naive reader), is that it is permissible to have non-uniform control flow executed after a call to this function.

One possibility is to rename the enums:

      SubsequentControlFlowMayBeNonUniform ->  MayCauseNonUniformControlFlow

      ReturnValueMayBeNonUniform -> MayReturnNonUniformValue

Either that, or change the description to say

     Calling this function may cause control flow to be nonuniform immediately after the call site.

I see there is some rhyming with the argument tags.  So maybe the best thing is to just update the description.